ui/src/components/RecordButton.js

```diff
@@ -40,116 +43,138 @@ export class RecordButton extends LitElement {
 
     constructor() {
         super();
+        this.disabled = false;
         this.audioRecorder = new AudioRecorder();
         this.utteranceService = new UtteranceApiService();
+        this.finishedAudio = undefined;
     }
 
-    updated() {
+    updated(changedProperties) {
```

Comment:
nit: do you need to include changedProperties if you're calling the functions regardless of the properties changed?